### PR TITLE
modules: cJSON: enable float printf support

### PR DESCRIPTION
**Pull request by eivindj-nordic**
cJSON library is using snprintf with floats. We therefore need to
enable NEWLIB_LIBC_FLOAT_PRINTF to ensure that these are handled
correctly.

Signed-off-by: Eivind Jølsgard <eivind.jolsgard@nordicsemi.no>

### CODE REVIEW

File: modules/cjson/Kconfig [hunk 1]
```diff
@@ -9,3 +9,6 @@ config CJSON_LIB
 	# Since minimal libc lacks sscanf and strtod we need to use newlibc
 	# Allow external libc to allow native posix to be used for unit test
 	depends on NEWLIB_LIBC || EXTERNAL_LIBC
+	# cJSON library is using snprintf with floats. We therefore need to
+	# enable NEWLIB_LIBC_FLOAT_PRINTF to ensure that these are handled correctly.
+	imply NEWLIB_LIBC_FLOAT_PRINTF if NEWLIB_LIBC
```
Comment (tejlmand) [inline modules/cjson/Kconfig:14]: > \# cJSON library is using snprintf with floats. We therefore need to
> \# enable NEWLIB_LIBC_FLOAT_PRINTF to ensure that these are handled correctly.

This sounds like `imply` is the wrong choice here, as `imply` allows users to disable `NEWLIB_LIBC_FLOAT_PRINTF`.

You already have a `depends on NEWLIB_LIBC` above, so should this be `depends on NEWLIB_LIBC_FLOAT_PRINTF` to be aligned.

**Review by tejlmand — CHANGES_REQUESTED** [review]
the use of `imply` doesn't seem correct according to the code comment a commit message.

Please explain.

File: modules/cjson/Kconfig (hunk 1 above)
Comment (eivindj-nordic) [reply modules/cjson/Kconfig:14]: Sounds reasonable, I will update the PR to `depends on`.

Comment (lemrey) [reply modules/cjson/Kconfig]: Probably need to enable this explicitly in the project configuration then

Comment (eivindj-nordic) [reply modules/cjson/Kconfig]: Yes, the plan was to add it to the `lte_ble_gateway` PR.

Comment (lemrey) [reply modules/cjson/Kconfig]: Both are fine

**Review by tejlmand — APPROVED** [review]
lgtm

**Review by simensrostad — APPROVED** [review]

**Comment by tejlmand** [comment]
@eivindj-nordic remember to update the change-log if this is fixing a bug.

File: doc/nrf/releases/release-notes-changelog.rst [hunk 2]
```diff
@@ -420,6 +420,11 @@ Trusted Firmware-M
 
 * Fixed the NCSDK-14015 known issue that would cause crash during boot when the :kconfig:`CONFIG_RPMSG_SERVICE` Kconfig option was enabled on the nRF5340 SoC.
 
+cJSON
+=====
+
+*  Fixed an issue with floats in cJSON module when using NEWLIB_LIBC without the :kconfig: `NEWLIB_LIBC_FLOAT_PRINTF` Kconfig option.
```
Comment (divipillai) [inline doc/nrf/releases/release-notes-changelog.rst:426]:
```suggestion
*  Fixed an issue with floats in the cJSON module when using NEWLIB_LIBC without the :kconfig:`NEWLIB_LIBC_FLOAT_PRINTF` Kconfig option.
```

**Review by divipillai — APPROVED** [review]

**Review by anhmolt — APPROVED** [review]

File: doc/nrf/releases/release-notes-changelog.rst [hunk 3]
```diff
@@ -420,6 +420,11 @@ Trusted Firmware-M
 
 * Fixed the NCSDK-14015 known issue that would cause crash during boot when the :kconfig:`CONFIG_RPMSG_SERVICE` Kconfig option was enabled on the nRF5340 SoC.
 
+cJSON
+=====
+
+*  Fixed an issue with floats in the cJSON module when using NEWLIB_LIBC without the :kconfig:`NEWLIB_LIBC_FLOAT_PRINTF` Kconfig option.
```
Comment (rlubos) [inline doc/nrf/releases/release-notes-changelog.rst:426]: Shouldn't this now be `:kconfig:option:`? CC @divipillai

**Review by rlubos — COMMENTED** [review]
Looks (almost) good, but please rebase your PR.

File: doc/nrf/releases/release-notes-changelog.rst (hunk 3 above)
Comment (divipillai) [reply doc/nrf/releases/release-notes-changelog.rst]: yes

Comment (eivindj-nordic) [reply doc/nrf/releases/release-notes-changelog.rst]: Thanks, updated it with the rebase :)

**Review by rlubos — APPROVED** [review]

**Pull request merged**